timemachine/cpp/src/kernels/k_neighborlist.cuh

```diff
@@ -303,12 +309,12 @@ void __global__ k_find_blocks_with_ixns(
         atom_box_dy = max(static_cast<RealType>(0.0), fabs(atom_box_dy) - col_bb_ext_y);
         atom_box_dz = max(static_cast<RealType>(0.0), fabs(atom_box_dz) - col_bb_ext_z);
 
+        bool check_column_atoms =
+            atom_i_idx < K &&
```

Comment:
This additional check is important, without this check you get junk interactions and the test fails.. Previously we were getting lucky with positions beyond the number of atoms we have.

Reply:
Good call!